crates/matrix-sdk-ui/src/timeline/inner/mod.rs

```diff
@@ -291,20 +291,34 @@ impl<P: RoomDataProvider> TimelineInner<P> {
         Ok(result)
     }
 
-    pub(super) async fn set_initial_user_receipt(
-        &mut self,
-        receipt_type: ReceiptType,
-        receipt: (OwnedEventId, Receipt),
-    ) {
+    pub(super) async fn populate_initial_user_receipt(&mut self, receipt_type: ReceiptType) {
         let own_user_id = self.room_data_provider.own_user_id().to_owned();
+
+        let mut read_receipt = self
+            .room_data_provider
+            .user_receipt(receipt_type.clone(), ReceiptThread::Unthreaded, &own_user_id)
+            .await;
+
+        // Fallback to the one in the main thread.
+        if read_receipt.is_none() {
+            read_receipt = self
+                .room_data_provider
+                .user_receipt(receipt_type.clone(), ReceiptThread::Main, &own_user_id)
+                .await;
+        }
```

Comment:
Can you add a test that if there's no unthreaded receipt, we do use the main-thread receipt instead, please?